Fix user guide (#150)
AY2223S1-CS2103T-T10-3/tp: docs/UserGuide.md

--------------------------------------------------------------------------------------------------------------------
## 1. Introduction to WorkBook

Are you a Computing student that finds it really hard to collate and manage all your internship progress together?
Do you find it really hard to get an internship because you do not know how to prepare for the different interview stages? Fret not,
WorkBook is just the right tool for you! <br><br>

> Welcome to WorkBook! :wave: <br><br>WorkBook is an **internship application tracker** that helps Computing students **prepare sufficiently** for their upcoming interviews to **secure** that internship.

Reviewer: ![Screenshot 2022-11-01 at 2 15 42 PM](https://user-images.githubusercontent.com/77665182/199170881-26061d5b-5d07-44b7-9c54-2a941b8b69ed.png)

would it be more readable if we insert a line break before "Fret not..."?
and also after adding this, the placement of "Welcome to WorkBook!..." is kind of awkward, maybe could move it to the top, or remove it?

4. Using your command terminal, navigate to the folder where you placed your WorkBook at and fire it up by inputting: `java -jar WorkBook.jar` in your terminal. <br> The GUI with sample data as shown below should appear in a few seconds.<br>
   ![Ui](images/Ui.png)

> 5. You can tell WorkBook what you want by typing your command in `Enter command here...` at the top of the application and pressing `enter` to execute. <br>

Reviewer 2: ```suggestion
5. You can tell WorkBook what you want by typing your command in `Enter command here...` at the top of the application and pressing <kbd>⏎ Enter</kbd> to execute. <br>
```

Reviewer 2: not sure if its a good idea but just thought it is interesting to try

   * **`exit`** : Exits the app.

> 6. You can refer to the [things to note](#3-things-to-note) below for details of each command.

Reviewer: description doesnt really match the actual content? (things to note is misc. info about commands, not details of each command)
maybe something like this instead?
```suggestion
6. You can jump straight to [here](#6-what-you-can-do) for a summary of the commands.
```
honestly can also remove it, not sure whats the use case for the hyperlink here

--------------------------------------------------------------------------------------------------------------------

> ## 3. About Our User Guide

Reviewer 2: ```suggestion
## 3. About this guide
```

This guide serves to:<br>

1. Educate new users on how to get started with WorkBook!
2. Explain the purpose and how to use each and every command!

> 3. Display a summary of all the commands you can utilize!

Reviewer: can probably save the exclamation marks for the top part when welcoming them, here seems abit excessive

--------------------------------------------------------------------------------------------------------------------
## 4. Exploring our User Guide

**Information Box**
<div markdown="block" class="alert alert-info">
**:information_source: Info:** Provides extra information that is useful
</div>

**Warning Box**
<div markdown="block" class="alert alert-warning">
**:exclamation: Warning: Important messages**
</div>

**Tip Box**
<div markdown="block" class="alert alert-success">
**:bulb: Tip:** Provides pointers to enhance your experience using the application 
</div>

> 

Reviewer 2: I would say these fall under things to note as well

Author: no

**Highlights** <br>
`commands` or `PARAMETERS`

**Keyboard Inputs**<br>

> <button>enter</button>   <button>&uarr;</button>   <button>&darr;</button>

Reviewer: are we actually using these though

Author: we are right?

Reviewer 2: I'd prefer not to copy wholesale, maybe classify it under: "Navigating the GUI"

</div>

### 6.1. Adding your internship application:

> Congratulations you have sent out a new internship application now you can,

Reviewer: ```suggestion
Congrats! You have already done the hard work - applying for that internship you've always wanted. Now, simply add the details into WorkBook and we can begin tracking your application!
```

add the application to WorkBook in [sorted order](#3-things-to-note).

Format: `add c/COMPANY r/ROLE s/STAGE [d/DATETIME] [e/COMPANY_EMAIL] [l/LANGUAGE TAG] [t/TAG]…​`

<div markdown="block" class="alert alert-success">
**:bulb: Tip:** The date and time you provide could represent multiple things:<br>

>   * The date and it happened

Reviewer: ```suggestion
  * The date and time it happened
```

  * The deadline of the corresponding `Stage`
</div>

Examples:
* `add c/Meta r/Frontend Engineer s/Application Sent d/29-Oct-2022 12:00 e/hrmonkey@example.com`
* `add c/Bytedance r/Backend Engineer s/Online Assessment d/24-Sep-2022 15:00 t/high pay l/Java l/Python`

### 6.2. Editing your internship application:

> In the scenario that your internship application has progressed to a different stage and requires updating this command,

Reviewer: ```suggestion
Good job! You've passed the Resume Screening and have been invited to complete an Online Assessment for your internship application. Now's a good time to use this command to update the application so that WorkBook can continue to give you relevant tips and reminders!
```

* You can remove all the internship’s tags by typing `t/` without
    specifying any tags after it.

<div markdown="block" class="alert alert-success">
**:bulb: Tip:** You can edit any number of attributes with the edit command!
</div>

Examples:
* `edit 1 s/Behavioural Interview e/hr@meta.com` Edits the stage and email address of the first internship to be `Behavioural Interview` and `hr@meta.com` respectively.
* `edit 2 l/golang t/` Adds `golang` as the only language tag for the second internship and clears all existing tags.

### 6.3. Listing all existing internship applications: 

> Lists all of your internship applications in WorkBook.

Reviewer: not sure if i recall correctly, are we moving the sorted stuff from Add into here?

Format: `list`

### 6.4. Finding your internships by company, stage or role name:

If you wanted to view all your internship applications corresponding to a particular company, stage or role,
then this command is for you! <br>

> It can find internships whose company, role or stage names contain any of the respective keywords.

Reviewer: it has to contain **all** of the keywords right?

then this command is for you! <br>

It can find internships whose company, role or stage names contain any of the respective keywords.

Format: `find c/COMPANY | r/ROLE | s/STAGE`

* Displays a list of internships that match all the keywords.

> * Case does not matter e.g. `meta` will match `Meta`.

Reviewer: is this not true?

> * Only full words will be matched e.g. `met` will not match `Meta`.

Reviewer: should remove from Tip if you are putting in Caution?

* Order of the keywords does not matter e.g. `Jane Street` will match `Street Jane`.
</div>

<div markdown="block" class="alert alert-warning">
**:exclamation: Caution**
* Only full words will be matched e.g. `met` will not match `Meta`.
* Exactly one attribute can be searched for either Company, Role or Stage.
</div>

Examples:
* `find r/Engineer` returns `Software Engineer` and `Backend Engineer`.
* `find s/Interview` returns `Technical Interview` and `Behavioural Interview`.

### 6.5. Deleting your internship application:

If you wanted to remove an internship application then this command,<br>

> Deletes the specified internship application from WorkBook.

Reviewer: not a fan of this line break here and then the capital letter on the next line

<img width="649" alt="Screenshot 2022-11-01 at 2 47 46 PM" src="https://user-images.githubusercontent.com/77665182/199175408-9f7c288d-9d6d-4961-8467-c11182c5a854.png">

* You can remove all the internship’s tags by typing `t/` without
    specifying any tags after it.

<div markdown="block" class="alert alert-success">

> **:bulb: Tip:** You can edit any number of attributes with the edit command!

Reviewer 2: no need exclamation mark

</div>

Examples:
* `edit 1 s/Behavioural Interview e/hr@meta.com` Edits the stage and email address of the first internship to be `Behavioural Interview` and `hr@meta.com` respectively.
* `edit 2 l/golang t/` Adds `golang` as the only language tag for the second internship and clears all existing tags.

### 6.3. Listing all existing internship applications: 

Lists all of your internship applications in WorkBook in a [sorted order](#3-things-to-note).

<div markdown="block" class="alert alert-success">
**:bulb: Tip:** Your most recent upcoming internship application will be at the top of the list!
</div>

Format: `list`

> ### 6.4. Finding your internships by company, stage or role name:

Reviewer 2: ```suggestion
### 6.4. Finding your internships applications
```

If you wanted to view all your internship applications corresponding to a particular company, stage or role,
then this command is for you! <br>

It can find internships whose company, role or stage names contain all the respective keywords.

Format: `find c/COMPANY | r/ROLE | s/STAGE`

* Displays a list of internships that match all the keywords.

<div markdown="block" class="alert alert-success">
**:bulb: Tip:** <br>
* Case does not matter e.g. `meta` will match `Meta`.
* Order of the keywords does not matter e.g. `Jane Street` will match `Street Jane`.
</div>

<div markdown="block" class="alert alert-warning">
**:exclamation: Caution**
* Only full words will be matched e.g. `met` will not match `Meta`.
* Exactly one attribute can be searched for either Company, Role or Stage.
</div>

Examples:
* `find r/Engineer` returns `Software Engineer` and `Backend Engineer`.
* `find s/Interview` returns `Technical Interview` and `Behavioural Interview`.

### 6.5. Deleting your internship application:

> If you wanted to remove an internship application then this command

Reviewer 2: if you want to remove an internship application, then this command...

deletes the specified internship application from WorkBook.

Format: `delete INDEX`

* Deletes the internship application at the specified `INDEX`.
* The index refers to the index number shown in the displayed internship application list.
* The index **must be a positive integer** 1, 2, 3, …​

<div markdown="block" class="alert alert-warning">
**:exclamation: Caution**

> * The command will only be executed if the index provided corresponds to an internship application

Reviewer 2: fullstop at the end

</div>

Examples:
* `list` followed by `delete 2` deletes the 2nd internship application in WorkBook.
* `find Meta` followed by `delete 1` deletes the 1st internship application within the results of the `find` command.

### 6.6. Clearing your existing internship applications:

> It's a new cycle of summer internship applications and that means it's time for some spring-cleaning! 

Reviewer 2: Gotta consider winter too, and there are many other potential uses of `clear`, so I was thinking:

Be it clearing your WorkBook for a new cycle of internships, or to get rid of our sample data, this command allows you to start with a clean WorkBook, removing all previously saved internship applications.

Then you can follow up with a tip box saying that all data in json files are immediately cleared after this command, so undo if you made a mistake and they will be back again or smth along those lines

* The index refers to the index number shown in the displayed internship application list.
* The index **must be a positive integer** 1, 2, 3, …​

<div markdown="block" class="alert alert-warning">
**:exclamation: Caution**

> * The command will only be executed if the index provided corresponds to a internship application

Reviewer: ```suggestion
* The command will only be executed if the index provided corresponds to an internship application
```

</div>

Examples:
* `list` followed by `delete 2` deletes the 2nd internship application in WorkBook.
* `find Meta` followed by `delete 1` deletes the 1st internship application within the results of the `find` command.

### 6.6. Clearing your existing internship applications:

> Clears all internship applications in the WorkBook.

Reviewer: Maybe can add a user story here too? something like:
```suggestion
Its a new cycle of summer internship applications and that means its time for some spring cleaning! Use this command to clear all the previous applications you've saved in WorkBook!
```

Example:
* `clear` removes all internship applications in the WorkBook.

<div markdown="block" class="alert alert-success">

> **:bulb: Tip:** If you cleared your internship applications by mistake, fret not you can easily undo this!

Reviewer: 💯

Reviewer 2: fret not *as* you can easily undo this
